src/main/java/uk/gov/hmcts/reform/sscs/services/ccd/CreateCcdService.java

```diff
@@ -23,33 +26,77 @@
+                     RetryTemplate retryTemplate) {
         this.ccdRequestDetails = ccdRequestDetails;
         this.coreCaseDataApi = ccd;
         this.idamService = idamService;
         this.startEventCcdService = startEventCcdService;
+        this.searchCcdService = searchCcdService;
+        this.retryTemplate = retryTemplate;
     }
 
-    @Retryable
     public CaseDetails create(SscsCaseData caseData, IdamTokens idamTokens) {
-        return tryCreate(caseData, idamTokens);
+        log.info("*** case-loader *** Starting create case process with SC number {} and ccdID {} ...",
+            caseData.getCaseReference(), caseData.getCcdCaseId());
+        try {
+            retryTemplate.execute(getRetryCallback(caseData, idamTokens), getRecoveryCallback(caseData, idamTokens));
+        } catch (Throwable throwable) {
+            log.error("Recovery mechanism failed when creating case with SC {} and ccdID {}...",
+                caseData.getCaseReference(), caseData.getCcdCaseId());
+        }
+        return null;
     }
 
-    @Recover
-    @SuppressWarnings("PMD.UnusedPrivateMethod")
-    private CaseDetails requestNewTokensAndTryToCreateAgain(SscsCaseData caseData, IdamTokens idamTokens) {
-        log.info("*** case-loader *** Requesting new idam and s2s tokens");
-        idamTokens.setIdamOauth2Token(idamService.getIdamOauth2Token());
-        idamTokens.setServiceAuthorization(idamService.generateServiceAuthorization());
-        return tryCreate(caseData, idamTokens);
+    private RecoveryCallback<CaseDetails> getRecoveryCallback(SscsCaseData caseData, IdamTokens idamTokens) {
+        return context -> {
+            log.info("*** case-loader *** Recovery method called when creating case with SC number {} and ccdID {}...",
+                caseData.getCaseReference(), caseData.getCcdCaseId());
+            idamTokens.setIdamOauth2Token(idamService.getIdamOauth2Token());
+            idamTokens.setServiceAuthorization(idamService.generateServiceAuthorization());
+            List<CaseDetails> ccdCases = searchCcdService.searchCasesByScNumberAndCcdId(idamTokens, caseData);
+            if (ccdCases.isEmpty()) {
+                return createCaseInCcd(caseData, idamTokens);
+            }
+            return ccdCases.get(0);
+        };
     }
 
-    private CaseDetails tryCreate(SscsCaseData caseData, IdamTokens idamTokens) {
+    private RetryCallback<CaseDetails, ? extends Throwable> getRetryCallback(SscsCaseData caseData,
+                                                                             IdamTokens idamTokens) {
+        return (RetryCallback<CaseDetails, Throwable>) context -> {
+            log.info("*** case-loader *** create case with SC number {} and ccdID {} and retry number {}",
+                caseData.getCaseReference(), caseData.getCcdCaseId(), context.getRetryCount());
+            List<CaseDetails> ccdCases = Collections.emptyList();
+            if (context.getRetryCount() > 0) {
```

Comment:
What is the point of this if statement? If it hits this RetryCallback surely we always need to check if the case exists?